backend/api/pipeline.proto

```diff
@@ -151,14 +179,53 @@ message Pipeline {
   string description = 4;
 
   // Output. The input parameters for this pipeline.
+  // TODO(jingzhang36): replace this parameters field with the parameters field
+  // inside PipelineVersion when all usage of the former has been changed to use
+  // the latter.
   repeated Parameter parameters = 5;
 
   // The URL to the source of the pipeline. This is required when creating the
   // pipeine through CreatePipeline API.
+  // TODO(jingzhang36): replace this url field with the code_source_urls field
+  // inside PipelineVersion when all usage of the former has been changed to use
+  // the latter.
   Url url = 7;
 
   // In case any error happens retrieving a pipeline field, only pipeline ID
   // and the error message is returned. Client has the flexibility of choosing
   // how to handle error. This is especially useful during listing call.
   string error = 6;
+
+  // Output only. The default version of the pipeline. As of now, the latest
+  // version is used as default. (In the future, if desired by customers, we
+  // can allow them to set default version.)
+  // TODO(jingzhang36): expose this in API pipeline definition with FE changes.
+  // PipelineVersion default_version = 8;
 }
+
+message PipelineVersion {
+  // Output. Unique version ID. Generated by API server.
+  string id = 1;
+
+  // Optional input field. Version name provided by user.
+  string name = 2;
+
+  // Output. The time this pipeline version is created.
+  google.protobuf.Timestamp created_at = 3;
+
+  // Output. The input parameters for this pipeline.
+  repeated Parameter parameters = 4;
+
+  // Input. Optional. Pipeline version code source.
+  string code_source_url = 5;
+
+  // Input. Required. Pipeline version package url.
```

Comment:
this should be optional. Same as the Url in here
https://github.com/kubeflow/pipelines/pull/1873/files#diff-84b54249a5b54df88b95c2eed5c3f7c3R187